setup(css-library): setup tailwind for all apps with global config in ui package

sidarth-23 opened the pull request:
… ui package

<!--
  Thanks for submitting a pull request!
  We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request. 
-->

## Summary

Made UI library a hub of any UI related things by all tailwind configs and css file required, an export by the ui library<

Also configured tailwind to the apps with the global config and is working
<!--
 Explain the **motivation** of this Pull Request
-->

- [ ✅ ] Tested
- [ ❌ ] Test Case added - Not applicable
- [ ✅ ] Build Successful
- [ ✅ ] Sufficient Code comments added
- [ ❌ ] Attached Screenshots / Videos - No screenshots for this
- [ ✅ ] All Relevant Documents added

## Depends on

No dependencies

<!--- Does this PR depend on another PR that should be merged first or at the same time -->

sidarth-23 committed:
setup(css-library): setup tailwind for all apps with global config in ui package

sidarth-23 committed:
Merge branch 'master' into setup/tailwind

yogai53 commented on packages/ui/tailwind.config.ts:
```diff
@@ -0,0 +1,17 @@
+import type { Config } from "tailwindcss";
+
```
Create new project for Tailwind.

sidarth-23 replied:
I'm done with it. But I'm unable to squash the commits due to git merge in between the commits

sidarth-23 committed:
setup(tailwind): created separate tailwind package and used it in all dependencies

setup(tailwind): bug fix in postcss

setup(tailwind): fix pnpm lock file

removed unused dependencies

yogai53 commented on packages/tailwind-config/package.json:
```diff
@@ -0,0 +1,19 @@
+    "tailwindcss": "^3.4.1",
+    "postcss": "^8.4.33",
+    "autoprefixer": "^10.4.17",
+    "typescript": "^5.3.3"
```
Why do we need this?

yogai53 replied:
Please check other dependencies and ensure if it is really needed for the project.

sidarth-23 replied:
I checked with the projects using it and found that it works only if tailwind is installed. But not sure if it's required in the config package. I'll test it out update the status here

sidarth-23 replied:
The intuition was correct and I did not need the node-modules in tailwind-config package. 

Except for the tailwindcss. I needed it to get the type of config file which is there in tailwindcss npm. Is it worth deleting tailwindcss from the config package and lose type safety for the config.ts file

sidarth-23 replied:
So we are adding in only the tailwindcss dependency for the config package

yogai53 committed:
Merge branch 'master' into setup/tailwind
